From abaafac10d5133c1a74a6211127991d97d65642d Mon Sep 17 00:00:00 2001
From: Jorge Cisneros <jorge.cisneros@hpe.com>
Date: Tue, 3 Jun 2025 20:56:52 +0000
Subject: [PATCH] fix: Use the nbd-client -g parameter to disable the use of
 the NBD_OPT_GO protocol message, and force the use of NBD_OPT_EXPORT_NAME
 instead.

Upstream-Status: Pending

Signed-off-by: Jorge Cisneros <jorge.cisneros@hpe.com>
---
 nbd-proxy.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/nbd-proxy.c b/nbd-proxy.c
index e6d12f3..0dcb7d0 100644
--- a/nbd-proxy.c
+++ b/nbd-proxy.c
@@ -157,7 +157,7 @@ static int start_nbd_client(struct ctx* ctx)
         dup2(fd, STDOUT_FILENO);
         close(fd);
 
-        execlp("nbd-client", "nbd-client", "-u", ctx->sock_path, "-n", "-L",
+        execlp("nbd-client", "nbd-client", "-u", ctx->sock_path, "-n", "-L", "-g",
                "-t", timeout_str, ctx->config->nbd_device, NULL);
         err(EXIT_FAILURE, "can't start ndb client");
     }
